File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.py
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        def traverse(node):
            if not node: return
            traverse(node.left)
            traverse(node.right)
            result.append(node.val)

        result = list()
        traverse(root)
        return result

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.cpp
class Solution {
    void traverse(TreeNode* node, vector<int>& result) {
        if (node == NULL) return;
        traverse(node->left, result);
        traverse(node->right, result);
        result.push_back(node->val);
    }

public:
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int> result;
        traverse(root, result);
        return result;
    }
};